fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java

```diff
@@ -0,0 +1,260 @@
+    }
+
+    public String keystoreType() {
+        return this.keystoreType.get();
+    }
+
+    public JoseConfiguration keystorePath(String path) {
+        this.keystorePath.set(path);
+        return this;
+    }
+
+    public String keystorePath() {
+        return this.keystorePath.get();
+    }
+
+    public JoseConfiguration signatureKeyAlias(String keyAlias) {
+        this.signatureKeyAlias.set(keyAlias);
+        return this;
+    }
+
+    public String signatureKeyAlias() {
+        return this.signatureKeyAlias.get();
+    }
+
+    public JoseConfiguration signatureAlgorithm(String algorithm) {
+        signatureAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String signatureAlgorithm() {
+        return this.signatureAlgorithm.get();
+    }
+
+    public JoseConfiguration signatureFormat(JoseFormat format) {
+        signatureFormat.set(format.name());
+        return this;
+    }
+
+    public JoseConfiguration signatureDataEncoding(boolean encoding) {
+        signatureDataEncoding.set(encoding);
+        return this;
+    }
+
+    public boolean signatureDataEncoding() {
+        return this.signatureDataEncoding.get();
+    }
+
+    public JoseConfiguration signatureDataDetached(boolean detached) {
+        signatureDataDetached.set(detached);
+        return this;
+    }
+
+    public boolean signatureDataDetached() {
+        return this.signatureDataDetached.get();
+    }
+
+    public JoseFormat signatureFormat() {
+        return JoseFormat.valueOf(this.signatureFormat.get());
+    }
+
+    public JoseConfiguration encryptionFormat(JoseFormat format) {
+        encryptionFormat.set(format.name());
+        return this;
+    }
+
+    public JoseFormat encryptionFormat() {
+        return JoseFormat.valueOf(this.encryptionFormat.get());
+    }
+
+    public JoseConfiguration encryptionKeyAlias(String keyAlias) {
+        this.encryptionKeyAlias.set(keyAlias);
+        return this;
+    }
+
+    public String encryptionKeyAlias() {
+        return this.encryptionKeyAlias.get();
+    }
+
+    public JoseConfiguration encryptionKeyPassword(String password) {
+        this.encryptionKeyPassword.set(password);
+        return this;
+    }
+
+    public String encryptionKeyPassword() {
+        return this.encryptionKeyPassword.get();
+    }
+
+    public JoseConfiguration keyEncryptionAlgorithm(String algorithm) {
+        keyEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String keyEncryptionAlgorithm() {
+        return this.keyEncryptionAlgorithm.get();
+    }
+
+    public JoseConfiguration contentEncryptionAlgorithm(String algorithm) {
+        contentEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String contentEncryptionAlgorithm() {
+        return this.contentEncryptionAlgorithm.get();
+    }
+    /**
+     * Keystore type.
+     */
+    @Configurable("swarm.jose.keystore.type")
+    @AttributeDocumentation("Keystore type")
```

Comment:
What are the possible values?

Reply:
Any type recognized by Java KeyStore (.jks is default) and additionally, "jwk" which is a collection of JWK keys

Reply:
I'll update the various docs a bit

Reply:
Thanks!